i18n4k-generator/src/main/kotlin/de/comahe/i18n4k/generator/I18n4kGenerator.kt

```diff
@@ -179,44 +153,132 @@ class I18n4kGenerator(
+                                                       .builder("createString")
+                                                       .maybeOverload()
+                                                       .addParams()
+                                                       .addLocale()
+                                                       .returns(String::class)
+                                                       .addStatement("return getStringN(messageIndex, %T(listOf($paramsKeyString), nameMapper), locale)",
+                                                                     ClassName("$pkg.messages.formatter", "MessageParametersList"))
+                                                       .build())
+                                      .addFunction(FunSpec
+                                                       .builder("createLocalizedString")
+                                                       .addParams()
+                                                       .returnLocalizedString()
+                                                       .addStatement("return getLocalizedString${params.size}(messageKey, messageIndex, $paramsKeyString, nameMapper)")
+                                                       .build())
+                                      .addFunction(FunSpec
+                                                       .builder("invoke")
+                                                       .maybeOverload()
+                                                       .addModifiers(KModifier.OPERATOR)
+                                                       .returns(String::class)
+                                                       .addParams()
+                                                       .addLocale()
+                                                       .addStatement("return createString($paramsKeyString, locale)")
+                                                       .build())
+                                      .addFunction(FunSpec
+                                                       .builder("get")
+                                                       .addModifiers(KModifier.OPERATOR)
+                                                       .addParams()
+                                                       .returnLocalizedString()
+                                                       .addStatement("return createLocalizedString($paramsKeyString)")
+                                                       .build())
+                                      .build())
+        }
+
         fieldNames.forEach { (key, fieldName) ->
             val params = bundle.getMessageParametersNames(key)
-                .filter { it != "~" }// remove the null token
-                .map { it.toString() }
-                .sortedWith(AlphanumComparator.INSTANCE_ENGLISH)
+                .filter { it.key != "~" }// remove the null token
+//                .toSortedMap(AlphanumComparator.INSTANCE_ENGLISH)
```

Comment:
Revert this. Sorting of the parameters is important! See also comment for `getMessageParametersNames`